### PR TITLE
feat: expire app data

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #492

- [x] This change addresses only certain aspects of the issue

## Description

This PR does the frontend changes for the ticket above which should remove access to application data for partners and jurisdictional admins 45 days after the listing closes. Does not include automated email notifications.

## How Can This Be Tested/Reviewed?

Login as a partner or jurisdictional admin and view the applications table page of a closed listing. You should see the message that notifies you of when the application data expires, which should be 45 days after the closing date. You can manually update closedAt to be more than 45 days in the past to see the empty state. An admin role experience should have no changes.

This changeset is an excellent candidate for a new unit test file, but I dug into trying to configure it and I think it might take a full day, and want to set aside that time later in the sprint to be sure we have capacity.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/pages/listings/[id]/applications/index.tsx [hunk 1]
```diff
@@ -160,86 +170,122 @@ const ApplicationsList = () => {
...
-                      onClick={() => setIsTermsOpen(true)}
-                      loadingMessage={csvExportLoading && t("t.formSubmitted")}
+      {shouldExpireData && expiryDate <= dayjs() ? (
+        <section className={pageStyles["expired"]}>
+          <div className={pageStyles["parent"]}>
+            <div className={pageStyles["container"]}>
+              <div className={pageStyles["main"]}>
+                <Card spacing={"lg"}>
+                  <CardSection>
+                    <Icon icon={faExclamationCircle} size="xl" />
```
Comment (mcgarrye) [inline sites/partners/src/pages/listings/[id]/applications/index.tsx:180]: Should this be hero icon?

**Comment by mcgarrye** [comment]
Screenshot 2024-05-08 at 3 26 56 PM
Screenshot 2024-05-08 at 3 27 01 PM
Seeing this signed in as jurisdictional admin

**Comment by emilyjablonski** [comment]
@mcgarrye Should be resolved sorry about that!

File: sites/partners/src/pages/listings/[id]/applications/index.tsx (hunk 1 above)
Comment (emilyjablonski) [reply sites/partners/src/pages/listings/[id]/applications/index.tsx:183]: The Hero icons haven't made their way into Doorway from core yet, and there's a Hero custom icon mapper added in that PR we should use

**Review by mcgarrye — APPROVED** [review]
LGTM

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**